sdks/java/core/src/main/java/org/apache/beam/sdk/runners/PTransformOverrideFactory.java

```diff
@@ -14,22 +14,28 @@
  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
  * See the License for the specific language governing permissions and
  * limitations under the License.
+ *
  */
-package org.apache.beam.runners.direct;
 
+package org.apache.beam.sdk.runners;
+
+import org.apache.beam.sdk.annotations.Experimental;
+import org.apache.beam.sdk.annotations.Experimental.Kind;
 import org.apache.beam.sdk.transforms.PTransform;
 import org.apache.beam.sdk.values.PInput;
 import org.apache.beam.sdk.values.POutput;
 
-interface PTransformOverrideFactory<
+/**
+ * Produces {@link PipelineRunner}-specific overrides of {@link PTransform PTransforms}, and
+ * provides mappings between original and replacement outputs.
+ */
+@Experimental(Kind.UNSPECIFIED)
```

Comment:
Idea: `Kind.CORE_RUNNERS_ONLY` as a blacklist for things that are unstable, for use only by our runners and other runner authors (much less pipeline authors) use at their own risk.